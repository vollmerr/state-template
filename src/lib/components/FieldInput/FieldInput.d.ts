import * as React from 'react';

export interface FieldInputProps extends HTMLInputElement {
  /** Accessible indicator of related information */
  'aria-describedby'?: string;

  /** Accessible indicator for errors existing */
  'aria-invalid'?: string;

  /** Class names to attach to the field */
  className?: string;

  /** Disable the input */
  disabled?: boolean;

  /** Id of field */
  id?: string;

  /** Label for the field */
  label: string;

  /** Name of field */
  name: string;

  /** Called when the field is blurred */
  onBlur?: (string: value?) => void;

  /** Called when the field changes */
  onChange?: (string: value?) => void;

  /** Called when the field is focused */
  onFocus?: (string: value?) => void;

  /** HTML tag to render as */
  tag?: string | React.ReactNode;

  /** Placeholder to display when no value */
  placeholder?: string;

  /** Value of the field */
  value?: string;
}

declare class FieldInput extends React.Component<FieldInputProps, {}> {}

export default FieldInput;
