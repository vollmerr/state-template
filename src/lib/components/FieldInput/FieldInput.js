import React from 'react';
import T from 'prop-types';

import { isEmptyText } from '../../utils/validate';
import { withField } from '../Field';

// input field that applies state-template styling
export const FieldInput = (props) => {
  const {
    tag: Tag,
    ...rest
  } = props;

  return (
    <Tag
      data-test={'field--input'}
      className={'form-control field-input'}
      {...rest}
    />
  );
};

FieldInput.propTypes = {
  /** HTML tag to render as */
  tag: T.node,
};

FieldInput.defaultProps = {
  tag: 'input',
};

export default withField(isEmptyText)(FieldInput);
