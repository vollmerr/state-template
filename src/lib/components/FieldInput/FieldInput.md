```jsx
<ExampleForm form={'input'}>
  <FieldInput
    name={'input'}
    label={'Input'}
    helpText={'This is the `helpText` content...'}
    onChange={(e,v) => { console.log('changing...', e, v) }}
  />
  <FieldInput 
    required
    name={'requiredInput'}
    label={'Required Input'}
  />
  <FieldInput 
    name={'validationsInput'}
    label={'Input with Custom Validations'}
    validate={[
      (val) => val !== 'example' 
        ? 'Value must be `example`' 
        : undefined
    ]}
  />
  <FieldInput 
    disabled
    name={'disabledInput'}
    label={'Disabled Input'}
  />
  <FieldInput 
    name={'textareaInput'}
    label={'Textarea Input'}
    tag={'textarea'}
  />
</ExampleForm>
```
