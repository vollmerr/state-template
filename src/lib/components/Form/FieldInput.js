import React from 'react';

import FieldWrapper from './FieldWrapper';
import withField from './withField';

// input field that applies state-template styling
const FieldInput = (props) => {
  const { input, type, disabled } = props;
  const { name } = input;
  const Control = type === 'textarea' ? 'textarea' : 'input';

  return (
    <FieldWrapper {...props}>
      <Control type={type || 'text'} className="form-control" id={name} disabled={disabled} {...input} />
    </FieldWrapper>
  );
};

export default withField(FieldInput);
