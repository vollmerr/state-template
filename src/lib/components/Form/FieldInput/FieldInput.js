import React from 'react';
import T from 'prop-types';

import { isEmptyText } from '../../../utils/validate';

import FieldWrapper from '../FieldWrapper';
import withField from '../withField';

// input field that applies state-template styling
export const FieldInput = (props) => {
  const {
    input, type, disabled, tag: Tag, ...rest
  } = props;
  const { name } = input;

  return (
    <FieldWrapper data-test={'field-input'} disabled={disabled} {...rest}>
      <Tag
        id={name}
        type={type}
        className="form-control field-input"
        disabled={disabled}
        {...input}
      />
    </FieldWrapper>
  );
};

FieldInput.propTypes = {
  /** Input from redux-form's Field, attaches name, value, etc  */
  input: T.object.isRequired,
  /** HTML tag to render as */
  tag: T.oneOfType([T.func, T.string]),
  /** type to use */
  type: T.string,
  /** Disable the input */
  disabled: T.bool,
};

FieldInput.defaultProps = {
  tag: 'input',
  type: 'text',
  disabled: false,
};

const withReduxField = withField(isEmptyText);
export default withReduxField(FieldInput);
