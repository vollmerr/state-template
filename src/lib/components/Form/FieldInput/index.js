import React from 'react';
import T from 'prop-types';

import { isEmptyText } from '../../../utils/validate';

import FieldWrapper from '../FieldWrapper';
import withField from '../withField';

// input field that applies state-template styling
const FieldInput = (props) => {
  const { input, type, disabled } = props;
  const { name } = input;
  const Control = type === 'textarea' ? 'textarea' : 'input';

  return (
    <FieldWrapper {...props}>
      <Control type={type} className="form-control" id={name} disabled={disabled} {...input} />
    </FieldWrapper>
  );
};

FieldInput.propTypes = {
  /** Input from redux-form's Field, attaches name, value, etc  */
  input: T.object.isRequired,
  /** Type of input */
  type: T.oneOf(['text', 'textarea']),
  /** Disable the input */
  disabled: T.bool,
};

FieldInput.defaultProps = {
  type: 'text',
  disabled: false,
};

const withReduxField = withField(isEmptyText);
export default withReduxField(FieldInput);
