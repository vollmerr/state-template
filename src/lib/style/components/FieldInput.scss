@media screen {
  .disabled {
    .field__input {
      background: $disabled-bg;
    }
  }
}
