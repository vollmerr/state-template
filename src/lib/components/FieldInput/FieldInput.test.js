import React from 'react';
import { shallow } from 'enzyme';

import { FieldInput } from './FieldInput';

const props = {
  name: 'test-name',
  label: 'test-label',
  onChange: jest.fn(),
};

let wrapper;
describe('FieldInput', () => {
  beforeEach(() => {
    wrapper = shallow(<FieldInput {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom `tag`s', () => {
    wrapper.setProps({ tag: 'textarea' });
    expect(wrapper.find('textarea').length).toBe(1);
  });

  it('should pass addtional props', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.prop('disabled')).toBe(true);
  });
});
