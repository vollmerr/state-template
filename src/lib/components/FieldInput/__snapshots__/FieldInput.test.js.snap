// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput should render correctly 1`] = `
<input
  className="form-control field-input"
  data-test="field--input"
  label="test-label"
  name="test-name"
  onChange={[MockFunction]}
/>
`;
